crates/common/src/dotrain_add_order_lsp.rs

```diff
@@ -1,73 +1,97 @@
 use crate::add_order::ORDERBOOK_ORDER_ENTRYPOINTS;
-use crate::frontmatter::try_parse_frontmatter_rebinds;
 use crate::rainlang::parse_rainlang_on_fork;
+use alloy_primitives::Address;
 use dotrain::{
     error::{ComposeError, ErrorCode},
     types::ast::Problem,
-    RainDocument, Rebind,
+    Rebind,
 };
 use dotrain_lsp::{
     lsp_types::{CompletionItem, Hover, Position, TextDocumentItem},
     RainLanguageServices,
 };
 use once_cell::sync::Lazy;
+use std::collections::HashMap;
 
 /// static lang services instance
 /// meta store instance can be taken from this for shared access to a unfied meta store across
 /// all the dotrain usage in this crate
 pub static LANG_SERVICES: Lazy<RainLanguageServices> = Lazy::new(RainLanguageServices::default);
 
-pub struct DotrainAddOrderLsp {
-    text_document: TextDocumentItem,
-    frontmatter: String,
-    rebinds: Option<Vec<Rebind>>,
+/// get hover for a given text document item
+pub fn hover(
```

Comment:
why did you rip out the struct impl? It's so much more legible to contain the functions in the context they'll be used.

Reply:
well I didnt see any reason why they are needed anymore, since all the args come from outside